code/PaymentGatewayController.php

```diff
@@ -14,22 +14,41 @@
  */
 class PaymentGatewayController extends \Controller
 {
-
-    private static $allowed_actions = array(
-        'endpoint'
-    );
+    private static $allowed_actions = [
+        'gateway'
+    ];
+    
+    private static $url_handlers = [
+        'gateway/$Gateway!/$Status/$Identifier' => 'gateway',
```

Comment:
I added it because I know some payment gateways allow adding of a variable to a "static" route. If I remember, WorldPay lets you do this on some of its URL's. I thought adding this might allow for greater flexibility...

Reply:
I see. We can leave this in there, but I'd probably rename it to avoid confusion with the actual payment identifier we use internally? Maybe name it `TransactionID` or something? Passing this to the extend-calls is a good idea though.

Reply:
I just removed it in the end, I think if you were using this sort of structure you would probably still use the standard URL structure.

I think you are right, it is just confusing really!